custom_dc/README.md

```diff
@@ -0,0 +1,227 @@
+# Custom Data Commons
+
+A custom Data Commons can hold new data and include UI modifications that are
+custom to a third party's need.
+
+This document illustrates development and deployment details for a custom Data
+Commons instance.
+
+## Prerequisites
+
+- Install [docker](https://www.docker.com/products/docker-desktop/)
+- [Optional] Install [gcloud](https://cloud.google.com/sdk/docs/install-sdk)
+  command line tool.
+
+## API Key
+
+- A custom Data Commons needs to connect with main Data Commons. Get API key for
+  Data Commons by sending an email to `support@datacommons.org`.
+
+- [Optional] Provision a Google Maps API key from your GCP project. This is
+  optional and used for place search in visualization tools. Refer to [Maps API
+  Key](TODO) section for detailed instructions.
+
+## Prepare Custom Data
+
+[Note] Skip this section if you don't have custom Data.
+
+Prepare CSV files with statistical data in the following formats:
+
+```csv
+geoId,observationDate,stat_var_1,var_2
+06,2021,555,666
+08,2021,10,10
+```
+
+```csv
+name,observationDate,stat_var_1,stat_var_2
+California,2021,555,666
+Colorado State,2021,10,10
+```
+
+```csv
+dcid,observationDate,stat_var_1,stat_var_2
+geoId/06,2021,555,666
+geoId/08,2021,10,10
+```
+
+The first column header is a property that identifies the observed entity,
+supported properties are `dcid`, `name`, `geoId`. When `dcid` is used, the
+entity should have been resolved (from a previous step).
+
+Refer to [sample folder](./sample) for supported example data files.
+
+Put all the input files under a local folder or Google Cloud Storage folder, and
+use the folder path for environment variable `SQL_DATA_PATH` as described in the
+Environment Variables section below.
+
+## Data Config
+
+A config file `config.json` is required and should be put under `SQL_DATA_PATH`.
+The detailed spec can be found in this
+[doc](https://github.com/datacommonsorg/import/blob/master/simple/stats/config.md).
+
+## UI Updates
+
+Default custom Data Commons use `custom` FLASK_ENV which uses a pre-defined html
+html and css customization. These customization are in the following folders:
+
+- html files: [location](../server/templates/custom_dc/custom/)
+- css and image files: [location](../static/custom_dc/custom/)
+
+Update these files for UI customization.
+
+## Local Development
+
+Custom Data Commons site via [localhost](http://localhost:8080) by following the
+instructions below:
+
+### Build docker image locally
+
+```bash
+docker build --tag datacommons-website-compose:latest \
+-f build/web_compose/Dockerfile \
+-t website-compose .
+```
+
+### Run custom Data Commons locally with SQLITE database
+
+In the root of this repository, run:
+
+```bash
+docker run -it \
+--env-file $PWD/custom_dc/sqlite_env.list \
+-p 8080:8080 \
+-v $PWD/custom_dc/sample:/sqlite \
+-v $PWD/server/templates/custom_dc:/workspace/server/templates/custom_dc \
```

Comment:
We had customizations that weren't just in /server/templates/custom_dc. Is there a reason not to mount the codebase from the root?

Reply:
@shifucun We've created new routes and pages, customized css, added topic page config files, customized templates, and customized some visualiztion tiles. Don't have the exact folders off the top of head but generally speaking, access to /server and /static should cover most of it I think

Reply:
the "static" folder mount won't work since they need to be compiled to be effective. Let me know which folders in "server" worth mounting and we can add to it.